File: joyal670-protenium-iro-hub-9d65cb9ae2c4/app/src/main/java/com/protenium/irohub/utils/MySelectorDecorator.java
package com.protenium.irohub.utils;

import android.app.Activity;
import android.graphics.drawable.Drawable;

import androidx.core.content.ContextCompat;

import com.prolificinteractive.materialcalendarview.CalendarDay;
import com.prolificinteractive.materialcalendarview.DayViewDecorator;
import com.prolificinteractive.materialcalendarview.DayViewFacade;
import com.protenium.irohub.R;

public class MySelectorDecorator implements DayViewDecorator {

    private final Drawable drawable;

    public MySelectorDecorator(Activity activity) {
        drawable = ContextCompat.getDrawable(activity, R.drawable.date_selector);
    }

    @Override
    public boolean shouldDecorate(CalendarDay day) {
        return true;
    }

    @Override
    public void decorate(DayViewFacade view) {
        view.setSelectionDrawable(drawable);

    }
}

File: proteinium/app/src/main/java/com/proteinium/proteiniumdietapp/utils/calendar/MySelectorDecorator.kt
package com.proteinium.proteiniumdietapp.utils.calendar

import android.app.Activity
import android.graphics.drawable.Drawable
import androidx.core.content.ContextCompat
import com.prolificinteractive.materialcalendarview.CalendarDay
import com.prolificinteractive.materialcalendarview.DayViewDecorator
import com.prolificinteractive.materialcalendarview.DayViewFacade
import com.proteinium.proteiniumdietapp.R

/**
 * Use a custom selector
 */
class MySelectorDecorator(context: Activity?) : DayViewDecorator {
    private val drawable: Drawable

    override fun shouldDecorate(day: CalendarDay): Boolean {
        return true
    }

    override fun decorate(view: DayViewFacade) {
        view.setSelectionDrawable(drawable)
    }

    init {
        drawable = ContextCompat.getDrawable(context!!, R.drawable.date_selector)!!
    }
}